pkg/infradb/infradb.go

```diff
@@ -1340,28 +1334,27 @@ func UpdateSviStatus(name string, resourceVersion string, notificationID string,
 		return nil
 	}
 
-	sviComponents := svi.Status.Components
-	for i, comp := range sviComponents {
-		compCounter := i + 1
-		if comp.Name == component.Name {
-			svi.Status.Components[i] = component
+	if component.Replay {
+		// One of the components has requested a replay of the DB.
+		// The task related to the status update will be dropped.
+		log.Printf("UpdateSviStatus(): Component %s has requested a replay\n", component.Name)
+		taskmanager.TaskMan.StatusUpdated(svi.Name, "svi", svi.ResourceVersion, notificationID, true, &component)
+		go startReplayProcedure(component.Name)
+		return nil
+	}
```

Comment:
the same about this block of code. It looks similar for each type

Reply:
Same answer as above.